compiler/z/codegen/OMRInstruction.hpp

```diff
@@ -88,6 +88,13 @@ class OMR_EXTENSIBLE Instruction : public OMR::Instruction
    int32_t getBlockIndex() { return _blockIndex; }
    void setBlockIndex(int32_t i) { _blockIndex = i; }
 
+   void setOpCodeValue(TR::InstOpCode::Mnemonic op)
+      {
+      OMR::Instruction::setOpCodeValue(op);
+
+      TR_ASSERT(_cg->getS390ProcessorInfo()->supportsArch(_opcode.getMinimumALS()), "Processor detected does not support instruction %s\n", TR::InstOpCode::getMnemonicName(op));
+      }
```

Comment:
@0xdaryl just making sure this will work the way I think it will. This function is not virtual but it should still "override" the base OMR implementation when being used in the Z codegen, correct?